examples/convert_eeg_to_bids.py

```diff
@@ -31,72 +38,107 @@
-# data format, we need to build a directory structure and supply meta data
-# files to properly *bidsify* this data.
+# The data are in the `European Data Format <https://www.edfplus.info/>`_ with
+# the ``.edf`` extension, which is good for us because next to the
+# `BrainVision format`_, EDF is one of the recommended file formats for EEG
+# data in BIDS format.
+#
+# However, apart from the data format, we need to build a directory structure
+# and supply meta data files to properly *bidsify* this data.
+#
+# We will do exactly that in the next step.
 
 ###############################################################################
-# Step 2: Formatting as BIDS
-# --------------------------
-#
-# Let's start by formatting a single subject. We are reading the data using
-# MNE-Python's io module and the :func:`mne.io.read_raw_edf` function. Note
-# that we must use `preload=False`, the default in MNE-Python. It prevents the
-# data from being loaded and modified when converting to BIDS.
-edf_path = eegbci.load_data(subject=1, runs=2)[0]
+# Convert to BIDS
+# ---------------
+#
+# Let's start with loading the data and extracting the events.
+# We are reading the data using MNE-Python's ``io`` module and the
+# :func:`mne.io.read_raw_edf` function.
+# Note that we must use the ``preload=False`` parameter, which is the default
+# in MNE-Python.
+# It prevents the data from being loaded and modified when converting to BIDS.
+
+# Load the data from "2 minutes eyes closed rest"
+edf_path = eegbci.load_data(subject=subject, runs=run)[0]
 raw = mne.io.read_raw_edf(edf_path, preload=False)
 
-###############################################################################
-# The annotations stored in the file must be read in separately and converted
-# into a 2D numpy array of events that is compatible with MNE.
-annot = mne.read_annotations(edf_path)
-raw.set_annotations(annot)
+# For converting the data to BIDS, we need to convert the the annotations
+# stored in the file to a 2D numpy array of events.
 events, event_id = mne.events_from_annotations(raw)
 
-print(raw)
+###############################################################################
+# For the sake of the example we will also pretend that we have the electrode
+# coordinates for the data recordings.
+# We will use a coordinates file from the MNE testing data in `CapTrak`_
+# format.
+#
+# .. note:: The ``*electrodes.tsv`` and ``*coordsystem.json`` files in BIDS are
+#           intended to carry information about digitized (i.e., *measured*)
+#           electrode positions on the scalp of the research subject. Do *not*
+#           (!) use these files to store "template" or "idealized" electrode
+#           positions, like those that can be obtained from
+#           :func:`mne.channels.make_standard_montage`!
+#
+
+# Get the electrode coordinates
+testing_data = mne.datasets.testing.data_path()
+captrak_path = os.path.join(testing_data, 'montage', 'captrak_coords.bvct')
```

Comment:
is this a "measured" montage or an "idealized" montage? I feel we should not show the users something bad and then say, "don't do this at home" :-)

There is already this example: https://2448-89170358-gh.circle-artifacts.com/0/dev/auto_examples/convert_ieeg_to_bids.html#sphx-glr-auto-examples-convert-ieeg-to-bids-py which shows how to attach montage and write them.

Reply:
it's a measured montage (If I remember correctly, even from my own head :wink: ).

in the iEEG example, the montage using [`make_dig_montage`](https://mne.tools/dev/generated/mne.channels.make_dig_montage.html#mne.channels.make_dig_montage), which is a way to get a montage from arrays ... here I show that MNE-Python also has readers for some common file formats --> `read_dig_captrak`

Reply:
@jasmainak please have a look at this again as well, it's ready to be merged :)

Reply:
okay fair enough